### PR TITLE
Add options validation feature documentation for preview1

### PR DESCRIPTION
**Pull request by HaoK**
@scottaddie @guardrex @Rick-Anderson idk what the official new process for how you want the dev docs for upcoming 2.2-preview1 features is, but here is my initial stab at this, let me know if you want me to provide more samples/words or whatever you need.

cc @ajcvickers

### CODE REVIEW

**Comment by guardrex** [comment]
Thanks @HaoK ... I'll add versioning and make a UE pass on it soon.

@scottaddie Can I simply add a 2.2 moniker range for the content?

**Comment by guardrex** [comment]
@HaoK See if I butchered the meaning of anything on the UE updates. These updates conform the text to our guidelines. This seems quite clear and easy to understand.

**Comment by guardrex** [comment]
@scottaddie This answers my earlier question. If I make a ritual sacrifice of three computer mice, do you think the :pray: *server gods* :pray: will give us the moniker now? :smile:

capture

**Comment by scottaddie** [comment]
@guardrex This is ready to use. Not sure if you saw the email or not.

**Comment by guardrex** [comment]
@scottaddie Yes, thanks. I'll make another pass and ping u for a quick UE review.

**Comment by HaoK** [comment]
Maybe just add clarification via example

- Validation of multiple options types: `class ValidateTwo : IValidateOptions<Option1>, IValidationOptions<Option2>`
- Validation that depends on another option type: `public DependsOnAnotherOptionValidator(IOptions<AnotherOption> options)`

File: aspnetcore/fundamentals/configuration/options.md [hunk 1]
```diff
@@ -243,6 +243,70 @@ named_options_2: option1 = ConfigureAll replacement value, option2 = 5
...
+   // e.OptionsName returns "optionalOptionsName"
+   // e.OptionsType returns typeof(MyOptions)
+   // e.Failures returns a list of errors, which would contain 
+   //     "custom error"
+}
+```
+
+The preceding example sets the named options instance to `optionalOptionsName`. The default options instance is `Options.DefaultName`.
+
+Validation runs when the options instance is created. Your option instance is guaranteed to pass validation the first time it's accessed.
```
Comment (scottaddie) [inline aspnetcore/fundamentals/configuration/options.md]: In the paragraph above, you use "options instance". Here, you use "option instance". Which one should it be?

Comment (HaoK) [reply aspnetcore/fundamentals/configuration/options.md]: I don't know whats the appropriate grammar, I am trying to refer to a specific named instance of the options type

Comment (guardrex) [reply aspnetcore/fundamentals/configuration/options.md]: Good :eye: ... "options" :+1:

Comment (guardrex) [reply aspnetcore/fundamentals/configuration/options.md]: The word is plural everywhere in the API (and hopefully the topic, too). Let's go with "options" for the technology and an "options instance" is a single options object.

**Comment by guardrex** [comment]
The topic was mostly consistent on "options." The last commit makes it fully compliant with that approach.

**Pull request merged**